app/models/service_terraform_template.rb

```diff
@@ -97,6 +97,17 @@ def job_option_key(action)
     "#{action.downcase}_job_options".to_sym
   end
 
+  def parse_dialog_options
+    dialog_options = options[:dialog] || {}
+
+    params = dialog_options.each_with_object({}) do |(attr, val), obj|
+      var_key = attr.sub(/^(password::)?dialog_/, '')
+      obj[var_key] = val unless var_key == attr
```

Comment:
Is this saying to _not_ clobber the value if it's already an extra var?  If so, I'm not sure that's right.  Typically the dialog overrides any hardcoded values.

Reply:
So background most of this method is pulled form embedded ansible, https://github.com/ManageIQ/manageiq/blob/master/app/models/service_ansible_playbook.rb#L167

I don't think that is the intent, since this is only looking at options[:dialog] and that isn't where hard-coded extra vars are stored.

I'm trying to dig into why ansible does this and the password decryption, we might be able to drop both and just go with a simpler approach.

Reply:
Ah okay I thought that was an ansible specific thing but protected dialog fields look like `"password::dialog_password"=>"v2:{1aCicMVgRQFLeGN5K/LWcA==}"}`

I'm not sure that I like where the decrypt is done with embedded_ansible because it looks like the Job.options will have the plaintext value...

Reply:
right, the password:: prefix is there to ensure that we don't print it in logs (if we see password in the key we are certain it's a sensitive value)

Reply:
This was added to embedded_ansible here https://github.com/ManageIQ/manageiq/pull/14636 and it is decrypted here https://github.com/ManageIQ/manageiq/pull/14636/files#diff-43965f59b3501c850937c667f9712b5d46a04da94da5385d74f2fc187f170ca0R16

I've opted to decrypt right before sending to Terraform::Runner so the plaintext values don't make it to the Job.options column.  I might be missing something on the embedded_ansible side maybe they're re-encrypting them before creating the Job but this seemed like a good way to go.

Reply:
That was also where the check for `unless var_key == attr` maybe so that password and non-password text boxes don't overwrite each other?